### ISSUE TITLE
OLM can't perform dependency resolution when depending on Tekton/OpenShift Pipelines operator

### ISSUE THREAD

**Issue by illeatmyhat** [tektoncd/operator#577]
# Expected Behavior
When creating an Operator which depends on this operator (either through the [CSV `requires` field](https://olm.operatorframework.io/docs/tasks/creating-operator-manifests/#required-apis), or through the [`dependencies.yaml`](https://docs.openshift.com/container-platform/4.9/operators/understanding/olm/olm-understanding-dependency-resolution.html#olm-bundle-format-dependencies_olm-understanding-dependency-resolution)), the OLM will determine the correct (usually latest) version of Tekton/OpenShift Pipelines to install, and add it to the Subscription's InstallPlan.
The OLM will then proceed to install all required resources.
# Actual Behavior
The OLM will fail to resolve dependencies, no InstallPlan is created, and the Subscription will hang, resulting in nothing being installed.
No obvious or consistent error message is produced.

# Steps to Reproduce the Problem
Create an Operator that depends on Tekton Pipelines, e.g in the `ClusterServiceVersion`:
```YAML
required:
- kind: TektonPipeline
  name: tektonpipelines.operator.tekton.dev
  version: v1alpha1
```
or
```YAML
required:
- kind: Pipeline
  name: pipelines.tekton.dev
  version: v1alpha1
```

# Additional Info
Debug advice: Check the pod logs in the `openshift-operator-lifecycle-manager` and `openshift-marketplace` namespaces, otherwise there is no obvious reason for Subscription installing to hang.

- Kubernetes version:

  **Output of `kubectl version`:**

 ```
Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.1-5-g76a04fc", GitCommit:"c66c03f", GitTreeState:"clean", BuildDate:"2021-02-03T17:45:18Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3+ffbb954", GitCommit:"3a0f2c90b43e6cffd07f57b5b78dd9f083e47ee2", GitTreeState:"clean", BuildDate:"2021-11-29T05:06:16Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"linux/amd64"}
 ```

- Tekton Pipeline version: N/A [Latest version available on OperatorHub is *1.6.1*]

**Comment by nikhil-thomas** [comment]
> Tekton Pipeline version: N/A [Latest version available on OperatorHub is 1.6.1]

I am assuming that you are on OpenShift 4.x. We have had other operators making the dependency work with the following in their CSV file.

```yaml
required:
      - description: Represents Installation of Tekton.
        displayName: OpenShift Pipelines Config
        kind: TektonConfig
        name: tektonconfigs.operator.tekton.dev
        version: v1alpha1
```

If the operator we are talking about is the one on OpenShift Platform, then you can get immediate help on, 
`openshift-pipelines` google chat or on `forum-tektoncd-pipelines` coreos slack.

**Comment by adambkaplan** [comment]
Shipwright's operator uses the approach above, and we can confirm that this works on upstream operatorhub.io
